android-sdk-simplexml/src/main/java/com/microsoft/hsg/android/simplexml/ConnectionFactory.java

```diff
@@ -38,108 +38,108 @@
-     */
-    private static Transport getTransport()
-    throws Exception
-    {
-        int connectTimeout = 30 * 1000;
-        int readTimeout = 30 * 1000;
-        String urlString =     HealthVaultApp.getInstance().getSettings()
-            .getServiceUrl();
-        
-        URL url = new URL(urlString);
-            
-        URLConnectionTransport transport = new URLConnectionTransport();
-        transport.setConnectionTimeout(connectTimeout);
-        transport.setReadTimeout(readTimeout);
-        transport.setUrl(url);
-        
-        return transport;
-    }
-        
-    /**
-     * Reset.
-     */
-    public static void reset() {
-        authenticator = null;
-    }
+	/** The authenticator. */
+	private static Authenticator authenticator;
+
+	/**
+	 * Gets the authenticator.
+	 *
+	 * @param appId the app id
+	 * @param authenticationSecret the authentication secret
+	 *
+	 * @return the authenticator
+	 *
+	 * @throws IOException Signals that an I/O exception has occurred.
+	 */
+	private static synchronized Authenticator getAuthenticator(
+			String appId, byte[] authenticationSecret)
+	throws IOException
+	{
+		if (authenticator == null) {
+			authenticator = new SodaAuthenticator(
+				appId, authenticationSecret);
+		}
+
+		return authenticator;
+	}
+
+	/**
+	 * Gets the unauthenticated connection.
+	 *
+	 * @return the unauthenticated connection
+	 */
+	public static Connection getUnauthenticatedConnection()
+	{
+		try
+		{
+			Connection connection = new Connection();
+			connection.setTransport(getTransport());
+
+			return connection;
+		}
+		catch (Exception e)
+		{
+			throw new HVException(e);
+		}
+	}
+
+	/**
+	 * Gets the connection.
+	 *
+	 * @param appId the app id
+	 * @param authenticationSecret the authentication secret
+	 *
+	 * @return the connection
+	 */
+	public static Connection getConnection(String appId, byte[] authenticationSecret)
+	{
+		try {
+			Connection connection = new Connection();
+			connection.setAppId(appId);
+			connection.setTransport(getTransport());
+			connection.setAuthenticator(getAuthenticator(
+					appId, authenticationSecret));
+			connection.authenticate();
+
+			return connection;
+		} catch (HVException hve) {
+			throw hve;
+		} catch (Exception e) {
+			throw new HVException(e);
+		}
+
+	}
+
+	/**
+	 * Gets the transport.
+	 *
+	 * @return the transport
+	 *
+	 * @throws Exception the exception
+	 */
+	private static Transport getTransport()
+	throws Exception
+	{
+		int connectTimeout = 30 * 1000;
+		int readTimeout = 30 * 1000;
+		String urlString =     HealthVaultApp.getInstance().getSettings()
+			.getServiceUrl();
+
+		URL url = new URL(urlString);
+
+		URLConnectionTransport transport = new URLConnectionTransport();
+		transport.setConnectionTimeout(connectTimeout);
+		transport.setReadTimeout(readTimeout);
+		transport.setUrl(url);
+
+		return transport;
+	}
+
+	/**
+	 * Reset.
+	 */
+	public static void reset() {
+		authenticator = null;
```

Comment:
Do we need to do anything with the connection when we reset? Does it disconnect automatically or will we need to call some kind of disconnect?